packages/block-library/src/image/image.js

```diff
@@ -788,14 +793,23 @@ export default function Image( {
 				which causes duplicated image upload. */ }
 			{ ! temporaryURL && controls }
 			{ img }
-			<Caption
-				attributes={ attributes }
-				setAttributes={ setAttributes }
-				isSelected={ isSingleSelected }
-				insertBlocksAfter={ insertBlocksAfter }
-				label={ __( 'Image caption text' ) }
-				showToolbarButton={ isSingleSelected && hasNonContentControls }
-			/>
+
+			{ lockCaption && (
+				<figcaption>{ attributes.caption?.toHTMLString() }</figcaption>
```

Comment:
Any HTML will be escaped by React (try using formatting or adding a link to the caption).

Is there another approach?

Reply:
TIL about `.toHTMLString()` 🙂 . The only issue is that if the caption text in the pattern is anything other than plain text, the HTML tags will show up in the caption in the editor: 

https://github.com/WordPress/gutenberg/assets/5417266/60627e24-578c-42f3-b393-1d06f7aa8f89

Could we pass it `dangerouslySetHTML` and escape it like: 

```diff
diff --git a/packages/block-library/src/image/image.js b/packages/block-library/src/image/image.js
index 4b46d085a77..3405c21007d 100644
--- a/packages/block-library/src/image/image.js
+++ b/packages/block-library/src/image/image.js
@@ -35,6 +35,7 @@ import { switchToBlockType } from '@wordpress/blocks';
 import { crop, overlayText, upload } from '@wordpress/icons';
 import { store as noticesStore } from '@wordpress/notices';
 import { store as coreStore } from '@wordpress/core-data';
+import { safeHTML } from '@wordpress/dom';
 
 /**
  * Internal dependencies
@@ -795,7 +796,11 @@ export default function Image( {
 			{ img }
 
 			{ lockCaption && (
-				<figcaption>{ attributes.caption?.toHTMLString() }</figcaption>
+				<figcaption
+					dangerouslySetInnerHTML={ {
+						__html: safeHTML( attributes.caption?.toHTMLString() ),
+					} }
+				></figcaption>
 			) }
 
 			{ ! lockCaption && (
```

I see that there's precedent for this in https://github.com/WordPress/gutenberg/pull/33814/files

Reply:
ah, yeh, html 🤦. Instead of using `dangerouslySetInnerHTML` we could pass through props to the RichText component to disable editing. We can do this either with an explicit `disableEditing` and use that to set the same `shouldDisableEditing` flag that block bindings `lockAttributesEditing` is using - [as here](https://github.com/WordPress/gutenberg/pull/58916/commits/7f33f240018353b34f0f508e1a3a4be486ee0811). 

Or we could pass  `contentEditable` and `aria-readonly`  props from the image caption and plumb these through the `Caption` component with a `...props`, [as here ](https://github.com/WordPress/gutenberg/pull/58916/commits/e1019b5eff991ec7d0b6cf4852e7ed901491f70c).

The second one does not alter the external API of the RichText component so maybe that is the approach to take?  Happy to hear others thoughts on this.

Reply:
> Or we could pass contentEditable and aria-readonly props from the image caption and plumb these through the Caption component with a ...props

It might be risky, as I think there's some programmatic modification of the `contenteditable` attribute directly via the DOM API elsewhere in the codebase (for example - https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/writing-flow/use-arrow-nav.js#L265).

So you could set it via React, but it might not stay that way.

> We can do this either with an explicit disableEditing and use that to set the same shouldDisableEditing flag that block bindings lockAttributesEditing is using

This could be a good idea, though it may need to be private initially, which means going through the dance of making a private version of the RichText component.

Reply:
yeh, I don't like relying on `...props` drilling for something critical as it can easily get broken somewhere in the component chain, as well as by the external DOM manipulations, so will follow up on the `disableEditing` approach even though it is trickier in terms of altering the API.